crates/exex/src/hints.rs

```diff
@@ -112,3 +123,294 @@ pub fn add_segment_hint() -> Hint {
+            // The signature payload
+            MaybeRelocatable::from(Felt252::from_bytes_be_slice(&value.to_bytes())),
+        ]
+        .into()
+    }
+}
+
+impl CairoSerializableBlock for SealedBlock {
+    fn to_cairo_vm_block_header(&self) -> KethPayload {
+        // Get the header from the block
+        let header = &self.header;
+
+        // Serialize the header fields into [`MaybeRelocatable`] elements
+        let serialized_header: Vec<KethPayload> = vec![
+            header.base_fee_per_gas.into(),
+            header.blob_gas_used.into(),
+            header.logs_bloom.into(),
+            header.beneficiary.into(),
+            header.difficulty.into(),
+            header.excess_blob_gas.into(),
+            header.extra_data.clone().into(),
+            header.gas_limit.into(),
+            header.gas_used.into(),
+            header.hash().into(),
+            header.mix_hash.into(),
+            header.nonce.into(),
+            header.number.into(),
+            header.parent_beacon_block_root.into(),
+            header.parent_hash.into(),
+            header.receipts_root.into(),
+            header.state_root.into(),
+            header.timestamp.into(),
+            header.transactions_root.into(),
+            header.ommers_hash.into(),
+            header.withdrawals_root.into(),
+        ];
+
+        // Flatten the serialized header into a single vector
+        KethPayload::from_iter(serialized_header.into_iter().flat_map(|field| field.0))
+    }
+
+    fn to_cairo_vm_block_body(&self) -> KethPayload {
+        // The first element is the length of the body (number of transactions)
+        let mut block_body: Vec<KethPayload> = vec![self.body.len().into()];
+
+        self.body.iter().for_each(|transaction| {
+            // RLP encode the transaction
+            let mut buf = Vec::new();
+            transaction.encode_without_signature(&mut buf);
+            // Add the transaction and its signature to the block body
+            block_body.extend([buf.into(), transaction.signature.into()]);
+        });
+
+        // Flatten the serialized block body into a single vector
+        KethPayload::from_iter(block_body.into_iter().flat_map(|field| field.0))
+    }
+}
+
+impl From<SealedBlock> for KethPayload {
+    fn from(block: SealedBlock) -> Self {
+        block.to_cairo_vm_block()
+    }
+}
+
+pub fn block_hint(block: SealedBlock) -> Hint {
+    Hint::new(
+        String::from("block"),
+        move |vm: &mut VirtualMachine,
+              _exec_scopes: &mut ExecutionScopes,
+              _ids_data: &HashMap<String, HintReference>,
+              _ap_tracking: &ApTracking,
+              _constants: &HashMap<String, Felt252>|
+              -> HintExecutionResult {
+            // This call will first add a new memory segment to the VM (the base)
+            // Then we load the block into the VM starting from the base
+            vm.gen_arg(&Into::<KethPayload>::into(block.clone()).0)?;
```

Comment:
where do you actually assign the base to the `block` cairo variable here?

Reply:
rust question: why do we need to clone?

Reply:
- For the base, this is done here no? https://github.com/lambdaclass/cairo-vm/blob/7d19956781901fb05c76f7c02fa82bd907f0f9a2/vm/src/vm/vm_memory/memory_segments.rs#L136 we add a new segment which is automatically set as the base for the block, is there anything else to set?
- For the clone, this is a closure question, for more explicitness, the exact error without cloning is `cannot move out of `block`, a captured variable in an `Fn` closure
move occurs because `block` has type `SealedBlock`, which does not implement the `Copy` trait`